File: Elevators/prototype/passenger.py

class Passenger:

    def __init__(self, start_floor_id, target_floor_id):
        self.start_floor_id = start_floor_id
        self.target_floor_id = target_floor_id
        self.time_spent_waiting = 0
        self.time_spent_travelling = 0

    def __repr__(self):
        return "[startFloorID={}, targetFloorID={}, timeSpentWaiting={}, timeSpentTravelling={}]".format(
            self.start_floor_id, self.target_floor_id, self.time_spent_waiting, self.time_spent_travelling
        )

File: Elevators/src/Passenger.cpp
/**
 * @file Passenger.cpp
 * @author Michael Schumacher, Tyler MacDonald
 * @brief Implementation file for the Passenger class
 * @version 0.1
 * @date 2021-11-28
 */

#include "Passenger.hpp"

std::string Passenger::toString() const {
    std::stringstream ss;
    ss << "[startFloorID=" << mStartFloorID;
    ss << ", targetFloorID=" << mTargetFloorID;
    ss << ", timeSpentWaiting=" << mTimeSpentWaiting;
    ss << ", timeSpentTravelling=" << mTimeSpentTravelling;
    ss << "]";
    return ss.str();
}

std::ostream& operator<< (std::ostream &os, const Passenger &s) {
    return (os << s.toString());
}
